### PR TITLE
Add config option files.gravity_tmp

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Add config option `files.gravity_tmp` to be used by `pihole -g` for the intermediate list files.

---

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

**Review by yubiuser — COMMENTED** [review]
Do we need migration code? To preserve the value from `setupVars.conf`?

File: src/config/cli.c [hunk 1]
```diff
@@ -481,13 +481,34 @@ int get_config_from_CLI(const char *key, const bool quiet)
...
+
+	// We first loop over all config options to check if the one we are
+	// looking for is an exact match, use partial match otherwise
+	bool exactMatch = false;
+	for(unsigned int i = 0; i < CONFIG_ELEMENTS; i++)
+	{
+		// Get pointer to memory location of this conf_item
+		struct conf_item *item = get_conf_item(&config, i);
+
+		// Check if item.k starts with key
```
Comment (yubiuser) [inline src/config/cli.c:493]:
```suggestion
		// Check if item.k is identical with key
```

File: src/config/cli.c [hunk 2]
```diff
@@ -481,13 +481,34 @@ int get_config_from_CLI(const char *key, const bool quiet)
...
+		if(strcmp(item->k, key) == 0)
+		{
+			exactMatch = true;
+			break;
+		}
+	}
+
+	// Loop over all config options again to find the one we are looking for
+	// (possibly partial match)
```
Comment (yubiuser) [inline src/config/cli.c:502]: Wouldn't it be easier to never start the loop if `exactMatch=true`

File: src/config/config.c [hunk 3]
```diff
@@ -1020,6 +1020,13 @@ void initConfig(struct config *conf)
 	conf->files.gravity.f = FLAG_ADVANCED_SETTING | FLAG_RESTART_FTL;
 	conf->files.gravity.d.s = (char*)"/etc/pihole/gravity.db";
 
+	conf->files.gravity_tmp.k = "files.gravity_tmp";
+	conf->files.gravity_tmp.h = "A temporary directory where Pi-hole can store files during gravity updates. This directory must be writable by the user running gravity (typically pihole).";
+	conf->files.gravity_tmp.a = cJSON_CreateStringReference("<any world-writable writable directory>");
```
Comment (yubiuser) [inline src/config/config.c:1025]:
```suggestion
	conf->files.gravity_tmp.a = cJSON_CreateStringReference("<any existing world-writable writable directory>");
```

File: src/config/cli.c (hunk 2 above)
Comment (DL6ER) [reply src/config/cli.c:502]: It'd increase code duplication which is why I didn't do this

**Review by yubiuser — APPROVED** [review]

**Pull request merged**